src/display/annotation_layer.js

```diff
@@ -1417,7 +1417,26 @@ class AnnotationLayer {
    * @memberof AnnotationLayer
    */
   static render(parameters) {
+    const sortedAnnotations = [];
+    let popupAnnotations = null;
+    // Ensure that Popup annotations are handled last, since they're dependant
+    // upon the parent annotation having already been rendered (please refer to
+    // the `PopupAnnotationElement.render` method); fixes issue 11362.
     for (const data of parameters.annotations) {
+      if (data.annotationType === AnnotationType.POPUP) {
+        if (!popupAnnotations) {
```

Comment:
I would just initialize this list above, just like `sortedAnnotations`, so we don't need this `if` statement. It's only one list and this method is not called often.

Reply:
Sure, that works too!